pkg/csi/service/vanilla/controller.go

```diff
@@ -115,53 +106,152 @@ func (c *controller) Init(config *cnsconfig.Config, version string) error {
+				c.manager.VcenterConfig.Host)
+			if err != nil {
+				log.Errorf("failed to verify if vSAN file services is supported or not. Error:%+v", err)
+				return err
+			}
+			if isvSANFileServicesSupported {
+				go common.ComputeFSEnabledClustersToDsMap(authMgr.(*common.AuthManager),
+					config.Global.CSIAuthCheckIntervalInMin)
+			}
+		}
+	} else {
+		// Multi vCenter feature enabled
+		c.managers = &common.Managers{
+			CnsConfig:      config,
+			VcenterManager: vcManager,
+		}
+		c.managers.VcenterConfigs = make(map[string]*cnsvsphere.VirtualCenterConfig)
+		c.managers.VolumeManagers = make(map[string]cnsvolume.Manager)
+		// Get VirtualCenterManager instance and validate version.
+		vcenterconfigs, err := cnsvsphere.GetVirtualCenterConfigs(ctx, config)
+		if err != nil {
+			log.Errorf("failed to get VirtualCenterConfigs. err=%v", err)
+			return err
+		}
+		for _, vcenterconfig := range vcenterconfigs {
+			vcenter, err := vcManager.RegisterVirtualCenter(ctx, vcenterconfig)
+			if err != nil {
+				log.Errorf("failed to register VC with virtualCenterManager. err=%v", err)
+				return err
+			}
+			c.managers.VcenterConfigs[vcenterconfig.Host] = vcenterconfig
+			c.managers.VolumeManagers[vcenterconfig.Host] = cnsvolume.GetManager(ctx, vcenter,
+				operationStore, idempotencyHandlingEnabled)
+		}
+		vCenters, err := common.GetVCenters(ctx, c.managers)
+		if err != nil {
+			log.Errorf("failed to get vcenter. err=%v", err)
+			return err
+		}
+		// Check vCenter API Version
+		for _, vc := range vCenters {
+			err = common.CheckAPI(ctx, vc.Client.ServiceContent.About.ApiVersion, common.MinSupportedVCenterMajor,
+				common.MinSupportedVCenterMinor, common.MinSupportedVCenterPatch)
+			if err != nil {
+				log.Errorf("checkAPI failed for vcenter API version: %s for vCenter %s, err=%v",
+					vc.Client.ServiceContent.About.ApiVersion, vc.Config.Host, err)
+				return err
+			}
+		}
+		if isAuthCheckFSSEnabled {
+			log.Info("CSIAuthCheck feature is enabled, loading AuthorizationService")
+			authMgrs, err := common.GetAuthorizationServices(ctx, vCenters)
+			if err != nil {
+				log.Errorf("failed to initialize authMgr. err=%v", err)
+				return err
+			}
+			c.authMgrs = authMgrs
+			for _, authMgr := range authMgrs {
+				go common.ComputeDatastoreMapForBlockVolumes(authMgr, config.Global.CSIAuthCheckIntervalInMin)
+			}
+			for _, vcconfig := range c.managers.VcenterConfigs {
+				isvSANFileServicesSupported, err := c.managers.VcenterManager.IsvSANFileServicesSupported(ctx,
+					vcconfig.Host)
+				if err != nil {
+					log.Errorf("failed to verify if vSAN file services is supported or not for vCenter: %s. "+
+						"Error:%+v", vcconfig.Host, err)
+					return err
+				}
+				if isvSANFileServicesSupported {
+					for _, authMgr := range authMgrs {
+						go common.ComputeFSEnabledClustersToDsMap(authMgr, config.Global.CSIAuthCheckIntervalInMin)
+					}
+				}
 			}
 		}
```

Comment:
Do you want to log and return an error if auth service is not enabled? If anyone tries to switch it off while using multi-VC, at least the controller will crash right?

Reply:
auth service feature has been there since many releases. I am planning to remove the feature gate and remove the option to disable auth service.